src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ClusterLeaderManager.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.ons.ssdc.caseprocessor.schedule;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.ClusterLeader;
+import uk.gov.ons.ssdc.caseprocessor.model.repository.ClusterLeaderRepository;
+
+@Component
+public class ClusterLeaderManager {
+  private static final Logger log = LoggerFactory.getLogger(ClusterLeaderManager.class);
+  private static final UUID LEADER_ID = UUID.fromString("e469807b-f2e2-47bd-acf6-74f8943ff3db");
```

Comment:
Unless I'm misunderstanding, this will need to change to generate a unique leader ID for each instance, otherwise they'll all have the same ID?

Reply:
we're using the DB as a locking mechanism, to ensure that only one pod has control of the row at one time, so that only one pod can become leader. It's by design that all the pods are attempting to lock the same row.

Reply:
Ah, I get it now thanks.